src/script/ceph-backport.sh

```diff
@@ -787,7 +787,7 @@ function maybe_deduce_remote {
     else
         assert_fail "bad remote_type ->$remote_type<- in maybe_deduce_remote"
     fi
-    remote=$(git remote -v | grep --extended-regexp --ignore-case '(://|@)github.com(/|:)'${url_component}'/ceph(\s|\.|\/)' | head -n1 | cut -f 1)
+    remote=$(git remote -v | grep --extended-regexp --ignore-case '(://+|@)github.com(/+|:|:/+)'${url_component}'/+ceph(\s|\.|\/)' | head -n1 | cut -f 1)
```

Comment:
in my case, i have 
```console
$ git remote -v
ci      git@github.com:ceph/ceph-ci.git (fetch)
ci      git@github.com:ceph/ceph-ci.git (push)
me      git@github.com:tchaikov/ceph.git (fetch)
me      git@github.com:tchaikov/ceph.git (push)
origin  git@github.com:ceph/ceph.git (fetch)
origin  git@github.com:ceph/ceph.git (push)
```
but if i add a git path with more slashes:
```console
$ git remote add origin-slash git@github.com://ceph/ceph.git
$ git fetch origin-slash main
fatal: protocol 'git@github.com' is not supported
$ git remote add origin-slash2 git@github.com:/ceph////ceph.git
$ git fetch origin-slash2 main
fatal: remote error:
  ceph////ceph is not a valid repository name
  Visit https://support.github.com/ for help
$ git remote add origin-slash3 https://///github.com/ceph/ceph.git
$ git fetch origin-slash3 main
fatal: unable to access 'https://///github.com/ceph/ceph.git/': URL using bad/illegal format or missing URL
$ git --version
git version 2.36.1
```
so it's also a legit to add have a single slash in the first component in a the git repo address with ssh protocol. but than one slash would not be accepted by git.

so based on my observation, probably we should use something like
```sh
    remote=$(git remote -v | grep --extended-regexp --ignore-case '(://|@)github.com(/|:|:/)'${url_component}'/ceph(\s|\.|\/)' | head -n1 | cut -f 1)
```